### PR TITLE
Prerequisits for Fedora

### PR DESCRIPTION
**Pull request by llrs**
Thanks to @Enchufa2 I got the dependencies in Fedora.

This PR is meant to merge after the windows one at #74

### CODE REVIEW

File: 02-getting_started.Rmd [hunk 1]
```diff
@@ -27,190 +27,315 @@ The `r-devel` at [R/trunk](https://svn.r-project.org/R/trunk "svn devel source c
...
+```sh
+sudo dnf install \
+    autoconf \
+    automake \
+    bzip2-devel \
+    cairo-devel \
+    flexiblas-devel \
+    gcc-c++ \
+    gcc-gfortran \
+    java-headless \
```
Comment (Enchufa2) [inline 02-getting_started.Rmd:102]: I've changed the building process, and this should be `java-devel` now, apologies.

File: 02-getting_started.Rmd [hunk 2]
```diff
@@ -27,190 +27,315 @@ The `r-devel` at [R/trunk](https://svn.r-project.org/R/trunk "svn devel source c
...
+    libtiff-devel \
+    libtool \
+    ncurses-devel \
+    pango-devel \
+    pcre2-devel \
+    readline-devel \
+    tcl-devel \
+    tex(inconsolata.sty) \
+    tex(latex) \
+    tex(upquote.sty) \
```
Comment (Enchufa2) [inline 02-getting_started.Rmd:122]: These need to be quoted, e.g. `'tex(latex)'`.

**Review by Enchufa2 — CHANGES_REQUESTED** [review]

File: 02-getting_started.Rmd [hunk 3]
```diff
@@ -27,190 +27,315 @@ The `r-devel` at [R/trunk](https://svn.r-project.org/R/trunk "svn devel source c
...
+      zlib1g-dev 
+```
+
+### Fedora
+
+<!-- # Command to obtain it: dnf rq -q --repo=fedora-source --requires R -->
+In Fedora you can use:
+
+```sh
+sudo dnf install \
```
Comment (Enchufa2) [inline 02-getting_started.Rmd:94]: You can use the `-y` flag here too if you wish an automatic "yes".

File: 02-getting_started.Rmd [hunk 4]
```diff
@@ -27,190 +27,315 @@ The `r-devel` at [R/trunk](https://svn.r-project.org/R/trunk "svn devel source c
...
+    libtool \
+    ncurses-devel \
+    pango-devel \
+    pcre2-devel \
+    readline-devel \
+    tcl-devel \
+    'tex(inconsolata.sty)' \
+    'tex(latex)' \
+    'tex(upquote.sty)' \
+    'texinfo-tex' \
```
Comment (Enchufa2) [inline 02-getting_started.Rmd:123]: Not this one. Sorry, I wasn't clear enough: only the ones that have parentheses require quoting.

**Comment by Enchufa2** [comment]
And the same applies to Ubuntu.

**Comment by Enchufa2** [comment]
R doesn't find the JDK unless we set `JAVA_HOME` for the configure script.

@llrs You could take the installation of the JDK (`default-jdk` for Ubuntu, `java-devel` for Fedora) to a separate optional step, as suggested by @mmaechler, and then add that, in such a case, the proper `JAVA_HOME` should be exported before running `./configure`, i.e.,

```
$ export JAVA_HOME=<path_to_jdk>
```

where this path is `/usr/lib/jvm/default-java` for Debian/Ubuntu, according to [Dirk's recipe](https://salsa.debian.org/edd/r-base/-/blob/master/debian/rules#L266), and `/usr/lib/jvm/jre` for Fedora, see [the spec](https://src.fedoraproject.org/rpms/R/blob/rawhide/f/R.spec#_356).

**Comment by AlexElledge** [comment]
The guide at https://cran.r-project.org/bin/linux/fedora/ says to keep R updated on Fedora using sudo dnf install R.

Using dnf to update R leaves me with R version 4.1.3

The current version appears to be 4.2.2

What is the issue here? How can I keep my R package up to date on Fedora?

**Comment by Enchufa2** [comment]
@AlexElledge Unfortunately we lagged behind updating R for Fedora 36 due to personal and organizational issues, so we have v4.1.3 there. Fedora 37 is launched this week with R 4.2.2.

**Comment by mmaechler** [comment]
I think this should be discussed more publicly on  R-SIG-Fedora (the dedicated mailing list):
  https://stat.ethz.ch/mailman/listinfo/r-sig-fedora

But as Inaki @Enchufa2 has replied already:
Note that we (all of Math + Stats at ETH Zurich) update our Fedora only once per academic year .. and only install the even numbers, i.e. F 28, 30, 32, 34, and now F 36 but then only next summer (academic break)  F 38. 
But then, we now  can run   `podman run -ti fedora`   and that will soon be F 37 ?

**Comment by Enchufa2** [comment]
> But then, we now can run `podman run -ti fedora` and that will soon be F 37 ?

Not exactly. Running this pulls `fedora:latest`, which is equivalent to `fedora:rawhide`, which will become F38. F37 was branched in August, and so you need to specify `podman run -ti fedora:37` to run this version.

And now that we are at it, when there's a mismatch between the R version in the current release of Fedora and rawhide, there's a straighforward way to run the latest R version (and any software in rawhide, for that matter) via toolbox, as explained [here](https://cran.r-project.org/bin/linux/fedora/#toolbox-container-based-development).

**Comment by AlexElledge** [comment]
@Enchufa2 thanks for the info! I'll be upgrading to 37 in a couple weeks so that's good to know. I don't really know what the process is for pushing updates to the official repository is, but if it's something you can train someone on fairly easily then I would volunteer to help in the future.

**Comment by llrs** [comment]
Thanks @mmaechler and @Enchufa2 for the feedback! I included rsync but I left the java dependency as it was to not complicate the process. 

I wait to merge #74 before merging this @hturner and @nbenn, but if you have any other feedback, the java installation/dependency or other, let me know.

**Comment by hturner** [comment]
I agree to leave the java dependency out for now, as our focus is on a straight-forward setup for people contributing to base R. So I've merged this in.

**Pull request merged**